File: data.structures/java/demos/queue/QueueDemo.java
/**
 * @author Pepe Gallardo, Data Structures, Grado en Informática. UMA.
 *
 * Simple Queue demo
 */

package demos.queue;

import dataStructures.queue.Queue;
import dataStructures.queue.ArrayQueue;
import dataStructures.queue.LinkedQueue;

public class QueueDemo {
	public static void main(String[] args) {
		Queue<Integer> q1 = new LinkedQueue<>();
		q1.enqueue(1);
		q1.enqueue(2);
		q1.enqueue(3);	
		int x = q1.first();
		q1.dequeue();
		
		Queue<Integer> q2 = new ArrayQueue<>();
		q2.enqueue(1);
		q2.enqueue(2);
		q2.enqueue(3);
		
		System.out.println(q1);
		System.out.println(q2);
	}
}

File: Recursos/data.structures/haskell/Demos/Queue/QueueDemo.hs
-------------------------------------------------------------------------------
-- Simple client module using a Queue
--
-- Data Structures. Grado en Informática. UMA.
-- Pepe Gallardo, 2012
-------------------------------------------------------------------------------

module Demos.Queue.QueueDemo where

import DataStructures.Queue.LinearQueue
import DataStructures.Queue.QueueAxioms(queueAxioms)


q1 :: Queue Int
q1 = enqueue 3 (enqueue 2 (enqueue 1 empty))

size :: Queue a -> Int
size s
  | isEmpty s =  0
  | otherwise =  1 + size (dequeue s)
